distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Case_Sensitivity.enso

```diff
@@ -31,12 +33,18 @@ type Case_Sensitivity
             TextFoldingStrategy.caseInsensitiveFold locale.java_locale
 
     ## PRIVATE
-       Is case insensitive when in memory.
+       Is case insensitive when in memory. This requires that, if
+       case-insensitive, that the locale be the default locale, and if it's not,
+       throws Illegal_Argument.
     is_case_insensitive_in_memory : Boolean
     is_case_insensitive_in_memory self = case self of
         Case_Sensitivity.Default -> False
         Case_Sensitivity.Sensitive -> False
-        Case_Sensitivity.Insensitive _ -> True
+        Case_Sensitivity.Insensitive locale -> case locale == Locale.default of
+            True -> True
+            False ->
+                msg = "Custom locales are not supported for regexes."
```

Comment:
I'm wondering if this message is good, if we ever decide to use this helper in a non-regex method.

For now it's ok, but it may be worth renaming it to `is_case_insensitive_for_regex` or changing the message to something more generic: `Custom locales are not supported for this operation.` (but ofc in general more detailed errors are better).

Reply:
Yes, this is what I was suggesting before; I agree with you that this method throws away the locale information, so it should always require it to be default. I'll make the error message more generic since it's a generic constraint.